src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IfError.cs

```diff
@@ -37,25 +38,56 @@ public IfErrorFunction()
                     new TypedName(ErrorType.ReifiedErrorTable(), new DName("AllErrors")),
                     new TypedName(DType.ObjNull, new DName("ErrorResult"))),
                 appliesToArgument: i => i > 0 && (i % 2 == 1));
+
+            // IfError(value1, fallback1, value2, fallback2, ..., valueN, [fallbackN], ...)
+            SignatureConstraint = new SignatureConstraint(omitStartIndex: 4, repeatSpan: 2, endNonRepeatCount: 0, repeatTopLength: 8);
         }
 
         public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
         {
             yield return new[] { TexlStrings.IfErrorArg1, TexlStrings.IfErrorArg2 };
-            yield return new[] { TexlStrings.IfErrorArg1, TexlStrings.IfErrorArg2, TexlStrings.IfErrorArg2 };
-            yield return new[] { TexlStrings.IfErrorArg1, TexlStrings.IfErrorArg2, TexlStrings.IfErrorArg2, TexlStrings.IfErrorArg2 };
         }
 
         public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
         {
             if (arity > 2)
             {
-                return GetGenericSignatures(arity, TexlStrings.IfErrorArg2);
+                return GetOverloadsIfError(arity);
             }
 
             return base.GetSignatures(arity);
         }
 
+        // Gets the overloads for the IfError function for the specified arity.
+        // If is special because it doesn't have a small number of overloads
+        // since its max arity is int.MaxSize.
+        private IEnumerable<TexlStrings.StringGetter[]> GetOverloadsIfError(int arity)
+        {
+            Contracts.Assert(arity >= 3);
+
+            // Limit the argCount avoiding potential OOM
+            var argCount = arity > SignatureConstraint.RepeatTopLength ? SignatureConstraint.RepeatTopLength + (arity & 1) : arity;
+            var signature = new TexlStrings.StringGetter[argCount];
+            var fOdd = (argCount & 1) != 0;
+            var cargCur = fOdd ? argCount - 1 : argCount;
+
+            for (var iarg = 0; iarg < cargCur; iarg += 2)
+            {
+                signature[iarg] = TexlStrings.IfErrorArg1;
+                signature[iarg + 1] = TexlStrings.IfErrorArg2;
+            }
+
+            if (fOdd)
+            {
+                signature[cargCur] = TexlStrings.IfErrorArg1;
+            }
+
+            argCount++;
+
+            var overloads = new List<TexlStrings.StringGetter[]> { signature };
+            return new ReadOnlyCollection<TexlStrings.StringGetter[]>(overloads);
```

Comment:
Do we need the extra construction of the ReadOnlyCollection here? Returning the list should be fine, I think?

Reply:
Simplified it by returning an array